lib/cors.js

```diff
@@ -1,21 +1,16 @@
-const defaults = require('defaults');
-const defined = require('defined');
-
+const { mergeRight } = require('omnibelt');
 const requiredHeaders = ['Accept', 'Content-Type', 'X-Amz-Date', 'Authorization', 'Accept-Version'];
 
 module.exports = {
   preflight: function(options) {
```

Comment:
At this point, this isn't really "options" so much as "standard args" - what if you just destructure here the things you need?
`function({ extraHeaders = [], allowMethods = [] }) {`

Also, you seem to be passing in a variable named `expectedHeaders` not `extraHeaders` now? 
Also, you are passing in `Set`s not Arrays, so you can't treat them like arrays in here?

Definitely seems like you need to test some options calls in the tests to make sure they are returning the headers you expect